### PR TITLE
Added more LoS load options

### LINKED ISSUE #103: BUG: datetime future warning
# Description
```
DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC)
```

## Test configuration
 - OS: All
 - Version: Python 3.12
 - Other details about your setup that could be relevant

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #103 and expands the options for loading line-of-site TEC data.

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality or documentation)
- This change requires a documentation update

# How Has This Been Tested?

Ran the example in the docs.  Tried various combinations of loading line of site data.

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.10
* Any details about your local setup that are relevant: develop branch of pysat

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysatMadrigal/instruments/madrigal_dst.py [hunk 1]
```diff
@@ -67,7 +67,8 @@
 madrigal_inst_code = 212
 madrigal_tag = {'': {'': "30006"}}
 madrigal_start = dt.datetime(1957, 1, 1)
-madrigal_end = dt.datetime.utcnow()
+madrigal_end = pysat.utils.time.filter_datetime_input(
```
Comment (rstoneback) [inline pysatMadrigal/instruments/madrigal_dst.py:70]: Drive-by quick comment: 
I'd recommend adding your rationale as a comment for filtering the date time to get one without a timezone, or with a timezone...

File: pysatMadrigal/instruments/madrigal_dst.py [hunk 2]
```diff
@@ -67,7 +67,8 @@
...
+    dt.datetime.now(tz=dt.timezone.utc))
```
Comment (aburrell) [inline pysatMadrigal/instruments/madrigal_dst.py]:
```suggestion
    dt.datetime.now(tz=dt.timezone.utc))  # Filters out timezone
```

File: pysatMadrigal/instruments/madrigal_geoind.py [hunk 3]
```diff
@@ -67,7 +67,8 @@
 madrigal_inst_code = 210
 madrigal_tag = {'': {'': "30007"}}
 madrigal_start = dt.datetime(1950, 1, 1)
-madrigal_end = dt.datetime.utcnow()
+madrigal_end = pysat.utils.time.filter_datetime_input(
+    dt.datetime.now(tz=dt.timezone.utc))
```
Comment (aburrell) [inline pysatMadrigal/instruments/madrigal_geoind.py]:
```suggestion
    dt.datetime.now(tz=dt.timezone.utc))  # Filters out timezone
```

File: pysatMadrigal/instruments/methods/general.py [hunk 4]
```diff
@@ -1102,7 +1103,8 @@ def get_remote_filenames(inst_code=None, kindat='', user=None, password=None,
         start = dt.datetime(1900, 1, 1)
 
     if stop is None:
-        stop = dt.datetime.utcnow()
+        stop = dt.datetime.now(tz=dt.timezone.utc)
+    stop = pysat.utils.time.filter_datetime_input(stop)
```
Comment (aburrell) [inline pysatMadrigal/instruments/methods/general.py:1109]:
```suggestion

    # Ensure the end time does not have timezone information
    stop = pysat.utils.time.filter_datetime_input(stop)
```

**Review by jklenzing — COMMENTED** [review]
Looks good. I would recommend adding the `if: ${{ always() }}` line to the "finish" job in the main workflow so that the final coverage step is run even if some of the env configs (eg, python 3.6.8) fail. This should at least be documented in an issue.

On a quasi-related note, I've been having issues getting coveralls to display the actual files when the coveralls app is used. :(

**Review by jklenzing — APPROVED** [review]
